File: Support/Templates/cs/Files/EditableRootBindingList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Templates
{
  [Serializable]
  public class EditableRootBindingList :
    BusinessBindingListBase<EditableRootBindingList, EditableChild>
  {
    private static void AddObjectAuthorizationRules()
    {
      // TODO: add authorization rules
      //AuthorizationRules.AllowGet(typeof(EditableRootBindingList), "Role");
    }

    [Fetch]
    private void Fetch(int criteria)
    {
      RaiseListChangedEvents = false;
      // TODO: load values into memory
      object childData = null;
      foreach (var item in (List<object>)childData)
        this.Add(DataPortal.FetchChild<EditableChild>(childData));
      RaiseListChangedEvents = true;
    }

    [Update]
    [Transactional(TransactionalTypes.TransactionScope)]
    private void Update()
    {
      // TODO: open database, update values
      //base.Child_Update();
    }
  }
}

File: Support/Templates/vb/Files/EditableRootBindingList.vb
Imports System
Imports System.Collections.Generic
Imports Csla
Imports Csla.Security

<Serializable()>
Public Class EditableRootBindingList
  Inherits BusinessListBase(Of EditableRootBindingList, EditableChild)

  Private Shared Sub AddObjectAuthorizationRules()
    ' TODO: add authorization rules 
    'Csla.Rules.BusinessRules.AddRule(GetType(EditableRootBindingList), New Csla.Rules.CommonRules.IsInRole(Rules.AuthorizationActions.GetObject, "Role"))
  End Sub

  <Fetch>
  Private Sub Fetch(ByVal criteria As Integer)
    RaiseListChangedEvents = False
    ' TODO: load values into memory 
    Dim childData As Object = Nothing
    For Each item As Object In DirectCast(childData, List(Of Object))
      Me.Add(DataPortal.FetchChild(Of EditableChild)(childData))
    Next
    RaiseListChangedEvents = True
  End Sub

  <Update>
  Private Sub Update()
    ' open database and update child objects
    'MyBase.Child_Update()
  End Sub

End Class
